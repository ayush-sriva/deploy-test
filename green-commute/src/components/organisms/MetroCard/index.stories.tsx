import { ComponentMeta, ComponentStory } from '@storybook/react';
import MetroCard from '.';

export default {
  title: 'components/Organisms/MetroCard',
  component: MetroCard,
} as ComponentMeta<typeof MetroCard>;

export const Template: ComponentStory<typeof MetroCard> = (args) => <MetroCard />;
