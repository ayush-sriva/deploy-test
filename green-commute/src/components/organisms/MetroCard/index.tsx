import { Grid, styled } from '@mui/material';
import Map from '../../../../public/assets/illustrations/mapview.png';
import TypographyComponent from '../../atoms/Typography';
import Illustration from '../../atoms/Illustration';
import ButtonComponent from '../../atoms/Button';
import responsiveTheme from '../../../theme';
import Eclipse from '../../../../public/assets/icons/eclipse.svg';
import Icon from '../../atoms/Icon';

const GridMainCOntainer = styled(Grid)({
  width: '25.25rem',
  height: '16.87rem',
  padding: '1.32rem',
});
const TypographyItem = styled(Grid)({
  color: `${responsiveTheme.palette.secondary.dark}`,
  paddingBottom: '0.3rem',
});
const RouteDetailsContainer = styled(Grid)({
  color: `${responsiveTheme.palette.secondary.darker}`,
  paddingBottom: '0.56rem',
});
const IconContainer = styled(Grid)({
  paddingLeft: '0.6rem',
  paddingRight: '0.6rem',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

const MetroCard = () => {
  return (
    <div>
      <GridMainCOntainer container direction='column'>
        <TypographyItem item>
          <TypographyComponent variant={'body2'} children={'catch a blue line metro towards Raidurg.'} />
        </TypographyItem>
        <Grid item>
          <RouteDetailsContainer container>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'â‚¹ 100'} />
            </Grid>
            <IconContainer item>
              <Icon url={Eclipse} />
            </IconContainer>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'25 Kms'} />
            </Grid>
            <IconContainer item>
              <Icon url={Eclipse} />
            </IconContainer>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'1 hr 20 min'} />
            </Grid>
          </RouteDetailsContainer>
        </Grid>
        <Grid item>
          <Illustration url={Map} />
        </Grid>
        <Grid item data-testid='Button1002'>
          <ButtonComponent
            children={'View in Google Maps'}
            buttonVariant={'text'}
            color={'primary'}
            size={'large'}
            typoVariant={'captionBold'}
          />
        </Grid>
      </GridMainCOntainer>
    </div>
  );
};

export default MetroCard;
