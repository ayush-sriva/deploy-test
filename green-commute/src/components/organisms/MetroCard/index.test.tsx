import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MetroCard from '.';

function renderMetroCard() {
  render(<MetroCard/>);
}

describe('CabCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MetroCard/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Text should render correctly', () => {
    renderMetroCard();
    const textElement = screen.getByText('catch a blue line metro towards Raidurg.');
    expect(textElement).toBeInTheDocument();
  });
  test('Text should render correctly', () => {
    renderMetroCard();
    const textElement = screen.getByAltText('illustration');
    expect(textElement).toBeInTheDocument();
  });
  test('Number of Icons should be 2', () => {
    renderMetroCard();
    const textElement = screen.getAllByAltText('Icon');
    console.log(textElement);
    expect(textElement.length).toBe(2);
  });
  test('Button element rendered correctly', () => {
    renderMetroCard();
    const ButtonElement = screen.getByTestId('Button1002');
    expect(ButtonElement).toBeInTheDocument();
  });
});
