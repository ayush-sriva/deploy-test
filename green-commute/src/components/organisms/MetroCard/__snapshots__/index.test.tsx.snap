// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabCard renders correctly 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1h24xol-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-1mezf00-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        catch a blue line metro towards Raidurg.
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1315b3h-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            â‚¹ 100
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-1t46eol-MuiGrid-root"
        >
          <img
            alt="Icon"
            data-testid="icon"
            src="eclipse.svg"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            25 Kms
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-1t46eol-MuiGrid-root"
        >
          <img
            alt="Icon"
            data-testid="icon"
            src="eclipse.svg"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            1 hr 20 min
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        alt="illustration"
        src="mapview.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      data-testid="Button1002"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-kpm0tx-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
        >
          View in Google Maps
        </span>
      </button>
    </div>
  </div>
</div>
`;
